scripts/segment_with_bio3d_geostas.R

```diff
@@ -0,0 +1,39 @@
+library(bio3d)
+library(jsonlite)
+
+args <- commandArgs(trailingOnly=TRUE)
+
+if (length(args) < 2) {
+  stop("USAGE: Rscript scripts/segment_with_bio3d_geostas.R <input.dcd> <output_dir>", call.=FALSE)
+}
+
+traj       <- read.dcd(args[1])
+output_dir <- args[2]
+
+dir.create(output_dir, recursive=T, showWarnings=F)
+
+write_clustering <- function(clustering, filename) {
+  clusters <- lapply(clustering$inds, function(ind) ind$atom)
+  json     <- toJSON(clusters, pretty=T)
+
+  f <- file(filename)
+  writeLines(json, f)
+  close(f)
+}
+
+clustering <- geostas(traj, k=2)
+
+# Reuse computed atomic movement similarity matrix (AMSM):
+amsm <- clustering$amsm
+
+write_clustering(clustering, paste0(output_dir, '/clustering02.json'))
+
+for (k in 3:9) {
+  clustering <- geostas(traj, amsm=amsm, k=k)
+  write_clustering(clustering, paste0(output_dir, '/clustering0', k, '.json'))
+}
+
+for (k in 10:10) {
+  clustering <- geostas(traj, amsm=amsm, k=k)
+  write_clustering(clustering, paste0(output_dir, '/clustering', k, '.json'))
+}
```

Comment:
Going from 2 to 10 is arbitrary -- which clustering is "best" is something I don't know if we can decide. However, different clusterings are very, very cheap once we have the AMSM (it would be *great* if we could put this into a file in a separate task). Having these as separate options in the TSV means having a dropdown in blender that lets the user decide which clustering to try.

GeoStaS also comes with a "hclust" clustering which I haven't tried yet.

The clustering is choppy -- there are short segments here and there. We could (and probably should) apply a "smoothing" where we connect short segments that are close enough and/or drop short segments that are too far from the bulk of the domain. But this kind of logic can be performed on the blender side as an option.